pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,138 @@
+# size returns the size of an attribute
+let size = \length
+    cond {
+        length = 'MAX': '',
+        //seq.has_prefix('0x', length): '(' ++ //seq.trim_prefix('0x', length) ++ ')',
+        length > 0: $`(${length})`,
+        _: '',
+    };
+
+# attributePatternToken returns the token based on attribute value
+let attributePatternToken = \entity \attr \token
+    cond {
+        (attr.options?:{}): token,
+        (entity.primary_key where //seq.contains(attr.name, .)): token,
+        let fks = (entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+            .reference_attribute orderby .)) where . != {}; (fks): token,
+        (attr.length = 'MAX'): token,
+        (//seq.has_prefix('0x', attr.length)): token,
+        _: '',
+    };
+
+# entityPatternToken returns the token based on entity value
+let entityPatternToken = \entity \model \token
+    cond {
+        (entity.primary_key | entity.foreign_keys | entity.cluster | model.indexes): token,
+        _: '',
+    };
+
+# sortingOrder determines and appends sorting order
+let sortingOrder = \e \attr
+    let pk = //seq.split('(', //seq.join('', (e.primary_key where //seq.contains(attr.name, .)) orderby .));
+    cond {
+        pk count > 1: ', ~' ++ //seq.trim_suffix(')', //str.lower(pk(1))),
+        _: '',
+    };
```

Comment:
Nit: This is just difficult to understand; the code may happen to work, but it's not really expressing its intent, so it's hard to know if it's right.

Reply:
ok